### PR TITLE
test: make state config mockable

### PR DESCRIPTION
**Pull request by pauhull**

### CODE REVIEW

File: internal/state/config/config.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+
+	toml "github.com/pelletier/go-toml/v2"
+)
+
+type Config interface {
+	Write() error
+	Marshal() ([]byte, error)
```
Comment (apricote) [inline internal/state/config/config.go:13]: This is only called from `Config.Write()` and can be removed from the interface.

**Review by jooola — APPROVED** [review]

File: internal/hcapi2/mock/mock_gen.go [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package hcapi2_mock
 
+//go:generate mockgen -package hcapi2_mock -destination zz_config_mock.go github.com/hetznercloud/cli/internal/state/config Config
```
Comment (apricote) [inline internal/hcapi2/mock/mock_gen.go:3]: I dont think this belongs into `internal/hcapi2/mock`. Its not a mock for `hcapi`. Perhaps `internal/state/config/mock`?

Comment (pauhull) [reply internal/hcapi2/mock/mock_gen.go]: Yes, I already realized that :D

File: internal/state/config/config.go (hunk 1 above)
Comment (pauhull) [reply internal/state/config/config.go:15]: What do you think about removing `ContextNames`, `ContextByName` and `RemoveContext` too? Should they stay mockable or can we just make a utility function that takes a `config.Config` and implements the functionality using `Contexts` and `SetContexts`?

Comment (apricote) [reply internal/state/config/config.go]: Fine by me, if it makes the code cleaner

Comment (pauhull) [reply internal/state/config/config.go]: Done in 6292032 and 68a7918

**Review by apricote — APPROVED** [review]

**Pull request merged**